src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OnSerializeTests.cs

```diff
@@ -419,10 +531,7 @@ public override void Write(Utf8JsonWriter writer, MyValue value, JsonSerializerO
         [Fact]
         public static void NonPocosIgnored()
         {
-            JsonSerializer.Serialize(new MyCollection());
```

Comment:
Could you also add a few tests in MetadataTests.cs that verify that callbacks can be specified via the `JsonTypeInfo` APIs?

Reply:
Tests added.